Controller/ElasticsearchController.php

```diff
@@ -612,6 +615,37 @@ public function searchApiAction(Request $request)
         ]);
     }
 
+    /**
+     * @param Request $request
+     *
+     * @param JobService $jobService
+     * @return RedirectResponse
+     * @Route("/search/export/{contentType}", name="emsco_search_export", methods={"POST"})
+     */
+    public function exportAction(Request $request, JobService $jobService, ContentType $contentType)
+    {
+        $exportDocuments = new ExportDocuments($contentType, $this->generateUrl('emsco_search_export', ['contentType' => $contentType]), '{}');
+        $form = $this->createForm(ExportDocumentsType::class, $exportDocuments);
+        $form->handleRequest($request);
+
+        /** @var ExportDocuments */
+        $exportDocuments = $form->getData();
+        $command = sprintf(
```

Comment:
Why run a command from the controller?? This is a strange behaviour. The controller and commands are interfaces to underlying services. Abstract the logic from the command to reuse it here. I think this apply to all jobs allready defined?

Reply:
That's how works ems asynchronous calls Todqy. I agree, it's an hack. I'm looking for a better solution. Nevertheless there is commit in this pull request which ones improve this functionality (flush output in DB on every writeln and infinite time limit).

Reply:
A message broker like redis should be considered

Reply:
Agree